org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/core/manipulation/OrganizeImportsOperation.java

```diff
@@ -323,6 +370,27 @@ public void add(SimpleName ref) {
 				}
 				typeBinding= typeBinding.getTypeDeclaration();
 				if (!typeBinding.isRecovered()) {
+					String qualifiedTypeName= typeBinding.getQualifiedName();
+					if (qualifiedTypeName.indexOf('.') > 0) {
+						String qualifier= qualifiedTypeName.substring(0, qualifiedTypeName.lastIndexOf('.'));
+						if (moduleImportsContains(qualifier)) {
+							for (Entry<String, Set<String>> entry : fOldModuleImports.entrySet()) {
+								if (entry.getValue().contains(qualifier)) {
```

Comment:
Just a random observation: the outer if seems redundant, given that we are doing the same iteration with contains check again.